### ISSUE TITLE
InvalidStateError: invalid state

### ISSUE THREAD

**Issue by razaqq** [gmarull/asyncqt#11]
Getting:
```
Traceback (most recent call last):
  File "site-packages\asyncqt\_windows.py", line 47, in _process_events
    f.set_result(value)
  File "C:\Users\User\AppData\Local\Programs\Python\Python37\lib\asyncio\windows_events.py", line 85, in set_result
    super().set_result(result)
asyncio.base_futures.InvalidStateError: invalid state
```

To fix it simply add a check in _windows.py on line 47:

```
    def _process_events(self, events):
        """Process events from proactor."""
        for f, callback, transferred, key, ov in events:
            try:
                self._logger.debug('Invoking event callback {}'.format(callback))
                value = callback(transferred, key, ov)
            except OSError:
                self._logger.warning('Event callback failed', exc_info=sys.exc_info())
            else:
                if not f.done():  # ADD THIS LINE
                    f.set_result(value)
```

**Comment by nocarryr** [comment]
I'm getting the same behavior as well.  The changes from PR #6 solved the issue.  #6 only checks if the Future was cancelled though.  Seems like `f.done()` would be safer (since it will return `True` if the Future was cancelled or done).

**Comment by gmarull** [comment]
Fixed by #6
